qiskit_machine_learning/neural_networks/circuit_qnn.py

```diff
@@ -191,18 +181,25 @@ def quantum_instance(self) -> QuantumInstance:
         return self._quantum_instance
 
     @quantum_instance.setter
-    def quantum_instance(self, quantum_instance) -> None:
+    def quantum_instance(
+        self, quantum_instance: Optional[Union[QuantumInstance, BaseBackend, Backend]]
+    ) -> None:
```

Comment:
@woodsp-ibm if the argument must be a valid `quantum_instance`, should we set `self._sampler` in the constructor? so:
```python
     self._quantum_instance = None
     self._sampler = None
     if quantum_instance is not None:
         self.quantum_instance = quantum_instance
```

Reply:
All instance variables should ideally be defined in the constructor - allows the class/instance vars to be quickly understood since its all in one place for readability. So yes they should be defined in the constructor - even when the code was like it was before. Often an IDE will highlight - mine does - if an instance var you are referring to is not defined in the constructor.

Reply:
@woodsp-ibm ok thanks. Will address shortly

Reply:
@stephenmurray2 Ok, we have a release next week that we want to have this in, so it really needs to be finished off this week which hopefully fits in with your 'shortly' timeframe.